indra/newview/llvovolume.cpp

```diff
@@ -5110,6 +5110,58 @@ LLControlAVBridge::LLControlAVBridge(LLDrawable* drawablep, LLViewerRegion* regi
 	mPartitionType = LLViewerRegion::PARTITION_CONTROL_AV;
 }
 
+void LLControlAVBridge::updateSpatialExtents()
+{
+	LL_PROFILE_ZONE_SCOPED_CATEGORY_DRAWABLE
+
+	LLControlAvatar* controlAvatar = getVObj()->getControlAvatar();
+
+	LLSpatialGroup* root = (LLSpatialGroup*)mOctree->getListener(0);
+
+	bool rootWasDirty = root->isDirty();
+	root->rebound(); // root becomes non-dirty here
+
+	// SL-18251 "On-screen animesh characters using pelvis offset animations
+	// disappear when root goes off-screen"
+	//
+	// Expand extents to include Control Avatar placed outside of the bounds
+	if (controlAvatar && (rootWasDirty || controlAvatar->mPlaying))
+	{
+		// Get coordinates of the Control Avatar bounding box
+		LLVector3 controlAvatarMin(controlAvatar->mDrawable->getSpatialExtents()[0].getF32ptr());
+		LLVector3 controlAvatarMax(controlAvatar->mDrawable->getSpatialExtents()[1].getF32ptr());
+
+		// Make coordinates relative to the root position
+		controlAvatarMin -= mDrawable->getPosition();
+		controlAvatarMax -= mDrawable->getPosition();
+
+		// Rotate coordinates backward to the root rotation
+		controlAvatarMin = controlAvatarMin * ~mDrawable->getRotation();
+		controlAvatarMax = controlAvatarMax * ~mDrawable->getRotation();
+
```

Comment:
When rotating an AABB, you can't rely on just the min/max corners to contain the original AABB in the new coordinate frame (for example, rotate an axis aligned cube min/max 45 degrees on Z and the min/max points will both be on the XY plane -- your new box will have zero volume).  You have to construct an OBB in the new coordinate frame and generate an AABB from that.